File: rb/features.rb
# Film SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module FilmFeatures
  def self.make_feature(name)
    case name
    when "base"
      FilmBaseFeature.new
    when "test"
      FilmTestFeature.new
    else
      FilmBaseFeature.new
    end
  end
end

File: py/features.py
# Film SDK feature factory

from feature.base_feature import FilmBaseFeature
from feature.test_feature import FilmTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FilmBaseFeature(),
        "test": lambda: FilmTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
